### PR TITLE
osd: remove invalid put on message

### PR DESCRIPTION
**Pull request by NitzanMordhai**
message converted to use smart pointer, the put is extra step that
causing the double put and valgrind error

Fixes: https://tracker.ceph.com/issues/52124
Signed-off-by: Nitzan Mordechai <nmordec@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/osd/PGBackend.cc [hunk 1]
```diff
@@ -157,8 +157,6 @@ void PGBackend::handle_recovery_delete(OpRequestRef op)
     [=](int r) {
       if (r != -EAGAIN) {
 	get_parent()->send_message_osd_cluster(reply, conn.get());
-      } else {
-	reply->put();
```
Comment (NitzanMordhai) [inline src/osd/PGBackend.cc:161]: the regression was introduced by https://github.com/ceph/ceph/commit/d4a71c3c5369c7a881822a6eb7624e3e80cc53de

**Review by tchaikov — APPROVED** [review]

File: src/osd/PGBackend.cc (hunk 1 above)
Comment (neha-ojha) [reply src/osd/PGBackend.cc:161]: you can add this information to the commit message

**Review by neha-ojha — APPROVED** [review]

**Comment by neha-ojha** [comment]
jenkins test make check

**Comment by NitzanMordhai** [comment]
jenkins test make check

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2022-08-07_05:39:26-rados-wip-kefu-testing-2022-08-07-1123-distro-default-smithi/
- https://pulpito.ceph.com/kchai-2022-08-07_07:44:14-rados-wip-kefu-testing-2022-08-07-1123-distro-default-smithi/

the `sudo yum -y install ceph-radosgw` failures are tracked by https://tracker.ceph.com/issues/57051

**Pull request merged**